chord_metadata_service/phenopackets/models.py

```diff
@@ -302,145 +201,255 @@ def get_project_id(self) -> Optional[str]:
     """
     Class to represent a statement about the contribution
     of a genomic element towards the observed phenotype
 
     FHIR: Observation
     """
-
+    UNKNOWN_STATUS = 'UNKNOWN_STATUS'
+    REJECTED = 'REJECTED'
+    CANDIDATE = 'CANDIDATE'
+    CONTRIBUTORY = 'CONTRIBUTORY'
+    CAUSATIVE = 'CAUSATIVE'
     GENOMIC_INTERPRETATION_STATUS = (
-        ('UNKNOWN', 'UNKNOWN'),
-        ('REJECTED', 'REJECTED'),
-        ('CANDIDATE', 'CANDIDATE'),
-        ('CAUSATIVE', 'CAUSATIVE')
-        )
-    status = models.CharField(max_length=200, choices=GENOMIC_INTERPRETATION_STATUS,
-                              help_text='How the call of this GenomicInterpretation was interpreted.')
-    gene = models.ForeignKey(Gene, on_delete=models.CASCADE,
-                             blank=True, null=True, help_text='The gene contributing to the diagnosis.')
-    variant = models.ForeignKey(Variant, on_delete=models.CASCADE,
-                                blank=True, null=True, help_text='The variant contributing to the diagnosis.')
+        (UNKNOWN_STATUS, 'Unknown status'),
+        (REJECTED, 'Rejected'),
+        (CANDIDATE, 'Candidate'),
+        (CONTRIBUTORY, 'Contributory'),
+        (CAUSATIVE, 'Causative')
+    )
+    subject_or_biosample_id = models.CharField(
+        max_length=200, blank=True, help_text="Id of the patient or biosample of the subject being interpreted")
+    interpretation_status = models.CharField(max_length=200, choices=GENOMIC_INTERPRETATION_STATUS,
+                                             default="UNKNOWN_STATUS",
+                                             help_text='How the call of this GenomicInterpretation was interpreted.')
+
+    # Corresponds to 'call' field in schema in case of GeneDescriptor
+    gene_descriptor = models.ForeignKey(GeneDescriptor, on_delete=models.CASCADE, null=True, blank=True,
+                                        help_text="Corresponds to 'call' field in schema in case of GeneDescriptor")
+    # Corresponds to 'call' field in schema in case of VariantInterpretation
+    variant_interpretation = models.ForeignKey(VariantInterpretation, on_delete=models.CASCADE, null=True, blank=True,
+                                               help_text="Corresponds to 'call' field in schema in case of "
+                                                         "VariantInterpretation")
+
     extra_properties = JSONField(blank=True, null=True,
                                  help_text='Extra properties that are not supported by current schema')
-    created = models.DateTimeField(auto_now_add=True)
-    updated = models.DateTimeField(auto_now=True)
 
     def clean(self):
-        if not (self.gene or self.variant):
+        if not (self.gene_descriptor or self.variant_interpretation):
             raise ValidationError('Either Gene or Variant must be specified')
 
     def __str__(self):
         return str(self.id)
 
 
-class Diagnosis(models.Model):
+class Diagnosis(BaseTimeStamp):
```

Comment:
now that this isn't foreign-keyd to a disease, I think it might need a cleanup function since it's many to many with genomicinterpretation

Reply:
`Diagnosis` is only an FK on `Interpretation`, so it should not need a cleanup? Unless you meant for `phenopacket.diseases`, which is a many-to-many?

Reply:
Oh nvm I see what you mean

Reply:
Interpretation & Diagnosis cleanup added